### PR TITLE
Vision Documentation Rewrite

### PR DESCRIPTION
**Pull request by Alphyte**
Added a section to Vision Processing. This section is meant to be read after the "Vision with Raspberry Pi" section.

### CODE REVIEW

**Comment by Daltz333** [comment]
My thoughts surrounding the vision processing section is to remove the majority of rio vision/target tracking documentation and having FRCVision being the primarily supported means of vision processing. Ideally I'd like to see that incorporated into this pull-request. 

Additionally, there already is a https://docs.wpilib.org/en/latest/docs/software/vision-processing/raspberry-pi/index.html section so that section should get merged with this.

**Comment by A1phyte** [comment]
I've deleted the following pages:

- Configuring Axis Camera
- Using the Microsoft Lifecam HD-3000
- Axis M1013 Camera Compatibility
- Using the Axis Camera at Single Network Events
- Using the CameraServer on the roboRIO

and merged the Vision with Raspberry Pi section into the FRCVision section. I've also removed all the references to the deleted pages in the Control Systems Hardware/Software pages.

**Comment by PeterJohnson** [comment]
There's really two things teams want to know how to do: (1) camera streaming to the DS, and (2) vision/image processing to find/aim at/drive to a vision target.  I think an introduction mapping out these two items and the various solutions and top level pros/cons for each (maybe even just a table?) would be helpful, with links to the pages with more detail on how to actually do it, and also organize the documentation heirarchy similarly.  (1) could be on the Rio, using FRCVision, or using an Axis camera.  (2) we could mention it can be done on the Rio, but we would guide folks into FRCVision, and then the flow you describe makes sense.  I think this would help separate out the Axis camera and Rio pieces as solutions for the specific need for only camera streaming.

**Comment by A1phyte** [comment]
@Daltz333 can you review again?

File: source/docs/software/vision-processing/frcvision/image-thresholding.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+By thresholding the image using HSV, you can seperate the image into the vision target (foreground), and the other things that the camera sees (background). The following code example converts a HSV image into a binary image by thresholding with HSV values.
+
+.. tabs::
+
+   .. code-tab:: py
+
+      binary_img = cv2.inRange(hsv_img, (min_hue, min_sat, min_val), (max_hue, max_sat, max_val))
+
+.. note::
+   These values may have to be tuned on an per-venue basis, as ambient lighting may differ across venues. It is recommended to allow editing of these values through NetworkTables in order to facilitate on-the-fly editing.
```
Comment (Daltz333) [inline source/docs/software/vision-processing/frcvision/image-thresholding.rst:44]: Just put this on the same line as ``.. note::``

File: source/docs/software/vision-processing/frcvision/using-cameraserver.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
+Using CameraServer
+==================
+
+Grabbing frames from CameraServer
```
Comment (Daltz333) [inline source/docs/software/vision-processing/frcvision/using-cameraserver.rst:4]: Use title case for all section titles

File: source/docs/software/vision-processing/frcvision/using-cameraserver.rst [hunk 3]
```diff
@@ -22,16 +22,16 @@ The FRCVision image comes with all the necessary libraries to make your own visi
...
-      input_img = np.zeros(shape=(width, height, 3), dtype=np.int8)
-
       while True:
          time, input_img = cvSink.grabFrame(input_img)
 
          if time == 0: # There is an error
             continue
-.. note::
-   OpenCV reads in the image as **BGR**, not **RGB** for historical reasons. Use ``cv2.cvtColor`` if you want to change it to RGB.
+.. note:: OpenCV reads in the image as **BGR**, not **RGB** for historical reasons. Use ``cv2.cvtColor`` if you want to change it to RGB.
```
Comment (Daltz333) [inline source/docs/software/vision-processing/frcvision/using-cameraserver.rst:30]: Need a newline between the codeblock and content

**Comment by Daltz333** [comment]
Merge conflicts :>

**Comment by Alphyte** [comment]
@PeterJohnson was told by Dalton to ping you for another review.

**Pull request merged**